pyartcd/pyartcd/pipelines/gen_assembly.py

```diff
@@ -0,0 +1,233 @@
+            cmd.append(",".join(self.arches))
+        cmd.append("release:gen-assembly")
+        cmd.append(f"--name={self.assembly}")
+        cmd.append("from-releases")
+        for nightly in candidate_nightlies:
+            cmd.append(f"--nightly={nightly}")
+        if self.custom:
+            cmd.append("--custom")
+        else:
+            if self.in_flight:
+                cmd.append(f"--in-flight={self.in_flight}")
+            for previous in self.previous_list:
+                cmd.append(f"--previous={previous}")
+            if self.auto_previous:
+                cmd.append("--auto-previous")
+        _, out, _ = await exectools.cmd_gather_async(cmd, stderr=None, env=self._doozer_env_vars)
+        return yaml.load(out)
+
+    async def _create_or_update_pull_request(self, fn: Callable[[OrderedDict], OrderedDict]):
+        """ Create or update pull request for ocp-build-data
+        :param fn: a callback function that makes changes to releases.yml
+        """
+        self._logger.info("Creating ocp-build-data PR...")
+        # Clone ocp-build-data
+        build_data_path = self._working_dir / "ocp-build-data-push"
+        build_data = GitRepository(build_data_path, dry_run=self.runtime.dry_run)
+        ocp_build_data_repo_push_url = self.runtime.config["build_config"]["ocp_build_data_repo_push_url"]
+        await build_data.setup(ocp_build_data_repo_push_url)
+        branch = f"auto-gen-assembly-{self.group}-{self.assembly}"
+        await build_data.fetch_switch_branch(branch, self.group)
+        # Load releases.yml
+        releases_yaml_path = build_data_path / "releases.yml"
+        releases_yaml = yaml.load(releases_yaml_path) if releases_yaml_path.exists() else {}
+        # Make changes
+        releases_yaml = fn(releases_yaml)
+        yaml.dump(releases_yaml, releases_yaml_path)
+        # Create a PR
+        title = f"Add assembly {self.assembly}"
+        body = f"Created by job run {self.runtime.get_job_run_url()}"
+        match = re.search(r"github\.com[:/](.+)/(.+)(?:.git)?", ocp_build_data_repo_push_url)
+        if not match:
+            raise ValueError(f"Couldn't create a pull request: {ocp_build_data_repo_push_url} is not a valid github repo")
+        head = f"{match[1]}:{branch}"
+        base = self.group
+        if self.runtime.dry_run:
+            self._logger.warning("[DRY RUN] Would have created pull-request with head '%s', base '%s' title '%s', body '%s'", head, base, title, body)
+            d = {"html_url": "https://github.example.com/foo/bar/pull/1234", "number": 1234}
+            result = namedtuple('pull_request', d.keys())(*d.values())
+            return result
+        pushed = await build_data.commit_push(f"{title}\n{body}")
+        result = None
+        if pushed:
+            github_token = os.environ.get('GITHUB_TOKEN')
+            if not github_token:
+                raise ValueError("GITHUB_TOKEN environment variable is required to create a pull request")
```

Comment:
If `DRY_RUN` is false, we do want to have a PR created. Then, moving this check at the very beginning of the pipeline could save us some time if things need to get fixed

Reply:
makes sense to move the check to the beginning.
In case of a dry run, `pushed` will be False so this will not be executed.